pkg/provisioner/ironic/ironic.go

```diff
@@ -55,7 +55,7 @@ const (
 	powerOff      = "power off"
 	softPowerOff  = "soft power off"
 	powerNone     = "None"
-	nameSeparator = "~"
+	nameSeparator = "-"
```

Comment:
This is [unsafe](https://github.com/metal3-io/baremetal-operator/pull/797#pullrequestreview-592348276); if we are going to change it, it must be to `.` not `-`.